app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,35 @@
+class VotesController < ApplicationController
+  before_action :authenticate_user!
+
+  def create
+    @quote = Quote.find params[:quote_id]
+    @vote = @quote.votes.new vote_params
+    @vote.user = current_user
+    if @vote.save
+      flash[:notice] = "Voted!"
+    else
+      flash.now[:alert] = "Something went wrong!"
+    end
+    redirect_to quotes_path
+  end
+
+  def update
+    @quote = Quote.find params[:quote_id]
```

Comment:
@quote = and @vote = at the beginning of every action can be extracted out to a before_action.